external/kafka/src/main/scala/org/apache/spark/streaming/kafka/KafkaUtils.scala

```diff
@@ -399,7 +418,7 @@ object KafkaUtils {
     val kc = new KafkaCluster(kafkaParams)
     val reset = kafkaParams.get("auto.offset.reset").map(_.toLowerCase)
 
-    (for {
+    KafkaCluster.checkErrors(for {
```

Comment:
What is this for? And the same above comment applies, reduce nesting by putting `for-yield` in separate val